repo/ipfs_keystore.go

```diff
@@ -58,12 +58,7 @@ func GetObjectFromIPFS(ctx commands.Context, p peer.ID, name string, maxCacheLen
 	getObjectFromIPFSCacheMu.Lock()
 	defer getObjectFromIPFSCacheMu.Unlock()
```

Comment:
This defer will release the lock before the goroutine below is necessarily done. You might get unreliable data.

Reply:
Not sure if this is a good time to use this or how much of a change it is: https://golang.org/pkg/sync/#Map. It's a map with the semaphore built in with O(n) lookup/storage.

Reply:
The function will continue and mutate the map after the lock has closed. https://play.golang.org/p/rHz6-LLFL4h

Reply:
But that's the behavior we want here isn't it?

Reply:
The lock is named `getObjectFromIPFSCacheMu`. It would seem it's trying to protect the map from being mutated. So I would guess not. But @tyler-smith helped me notice we have the lock inside the go routine also, which I overlooked. The lock inside the go routine should be enough.